segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +175,59 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> setupTokenExpiryTask(SetupAppend setupAppend) {
+        String segment = setupAppend.getSegment();
+        UUID writerId = setupAppend.getWriterId();
+        long requestId = setupAppend.getRequestId();
+
+        Duration expiryDuration = this.durationToExpiry(setupAppend);
+
+        if (expiryDuration == null) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.delayedTask(() -> {
+                if (isSetupAppendCompleted(segment, writerId)) {
+                    try {
+                        log.trace("Informing writer {} that sent request {}, about token expiry for segment {}",
+                                writerId, requestId, segment);
+                        if (!connection.isClosed()) {
+                            connection.send(new WireCommands.AuthTokenCheckFailed(setupAppend.getRequestId(),
+                                    String.format("Token sent by writer %s for segment %s in request %s has expired",
+                                            writerId, segment, requestId),
+                                    WireCommands.AuthTokenCheckFailed.ErrorCode.TOKEN_EXPIRED));
+                        }
+                    } catch (RuntimeException e) {
```

Comment:
RuntimeException extends Exception. Please catch Exception.

Reply:
I want to be specific here, as the intention is only to catch any runtime exceptions.